snuba/datasets/processors/search_issues_processor.py

```diff
@@ -0,0 +1,82 @@
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence, TypedDict
+
+from snuba import environment
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.datasets.events_format import EventTooOld, enforce_retention
+from snuba.datasets.processors import DatasetMessageProcessor
+from snuba.processor import InsertBatch, ProcessedMessage
+from snuba.utils.metrics.wrapper import MetricsWrapper
+
+metrics = MetricsWrapper(environment.metrics, "search_issues.processor")
+
+
+class SearchIssueMessage(TypedDict, total=False):
```

Comment:
Where does this get created? Currently, most event data is in the `data` key of the  message passed to kafka. I assumed we'd serialize the occurrence data and pass it via an `occurrence` key or something along those lines.

Reply:
https://github.com/getsentry/sentry/blob/f2c1f3648cb2000f12a4025b2e509ff16fbc8280/src/sentry/eventstream/snuba.py#L166-L197

Here's what the message looks like. Just not sure how that gets converted to this flattened message

Reply:
Yeah, we could go with that as well. I just have this class here as starting point to pull out data from the message. We can certainly update.

Reply:
Is the class here basically meant to be a reflection of what is sent via the Kafka message? Or is there a step where we translate it to this format?

Reply:
Yeah just a reference class so we know what fields are being sent. I made some changes to better reflect the messaging format we're more likely to be sending (based off the eventstream code).